File: September 2026/2026-09-02/2091/2091-removing-max-and-min-from-array.cpp
// 2091-removing-max-and-min-from-array.cpp;array;alone;leetcode
// https://leetcode.com/problems/removing-minimum-and-maximum-from-array/?envType=daily-question&envId=2026-09-02

class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();

        int minIndex = 0;
        int maxIndex = 0;

        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[minIndex]){
                minIndex = i;
            }

            if (nums[i] > nums[maxIndex]) {    
                maxIndex = i;
            }
        }

        int left = min(minIndex, maxIndex);
        int right = max(minIndex, maxIndex);

        int front = right + 1;
        int back = n - left;
        int both = (left + 1) + (n - right);

        return min({front, back, both});
    }
};

File: September 2026/2026-09-02/2091/2091-removing-max-and-min-from-array.py
# 2091-removing-max-and-min-from-array.py;array;alone;leetcode
# https://leetcode.com/problems/removing-minimum-and-maximum-from-array/?envType=daily-question&envId=2026-09-02
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)

        minI = nums.index(min(nums))
        maxI = nums.index(max(nums))

        left = min(minI, maxI)
        right = max(minI, maxI)

        front = right + 1
        back = n - left

        both = (left + 1) + (n - right)

        return min(front, back, both)